src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingController.java

```diff
@@ -0,0 +1,287 @@
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Pane;
+import javafx.scene.layout.RowConstraints;
+import javafx.scene.layout.StackPane;
+import javafx.scene.layout.VBox;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static javafx.beans.binding.Bindings.createBooleanBinding;
+import static javafx.beans.binding.Bindings.createObjectBinding;
+import static javafx.beans.binding.Bindings.createStringBinding;
+
+@Component
+@RequiredArgsConstructor
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+@Slf4j
+public class TeamMatchmakingController extends AbstractViewController<Node> {
+
+  private static final PseudoClass LEADER_PSEUDO_CLASS = PseudoClass.getPseudoClass("leader");
+  private static final PseudoClass CHAT_AT_BOTTOM_PSEUDO_CLASS = PseudoClass.getPseudoClass("bottom");
+
+  private final CountryFlagService countryFlagService;
+  private final AvatarService avatarService;
+  private final PlayerService playerService;
+  private final I18n i18n;
+  private final UiService uiService;
+  private final TeamMatchmakingService teamMatchmakingService;
+  private final FafService fafService;
+  private final EventBus eventBus;
+
+  public StackPane teamMatchmakingRoot;
+  public Button invitePlayerButton;
+  public Button leavePartyButton;
+  public Label refreshingLabel;
+  public ToggleButton uefButton;
+  public ToggleButton cybranButton;
+  public ToggleButton aeonButton;
+  public ToggleButton seraphimButton;
+  public ImageView avatarImageView;
+  public ImageView countryImageView;
+  public Label clanLabel;
+  public Label usernameLabel;
+  public Label gameCountLabel;
+  public Label leagueLabel;
+  public HBox queueBox;
+  public GridPane partyMemberPane;
+  public VBox preparationArea;
+  public ImageView leagueImageView;
+  public Label matchmakerHeadingLabel;
+  public Label partyHeadingLabel;
+  public Label queueHeadingLabel;
+  public ScrollPane scrollPane;
+  public HBox playerCard;
+  public Label crownLabel;
+  public TabPane chatTabPane;
+  public GridPane contentPane;
+  public ColumnConstraints column2;
+  public RowConstraints row2;
+  private Player player;
+  private MatchmakingChatController matchmakingChatController;
+
+  @Override
+  public void initialize() {
+    eventBus.register(this);
+    JavaFxUtil.fixScrollSpeed(scrollPane);
+    initializeDynamicChatPosition();
+
+    player = playerService.getCurrentPlayer().get();
+    initializeUppercaseText();
+    countryImageView.imageProperty().bind(createObjectBinding(() -> countryFlagService.loadCountryFlag(
+        StringUtils.isEmpty(player.getCountry()) ? "" : player.getCountry()).orElse(null), player.countryProperty()));
```

Comment:
Again here `StringUtils.isEmpty(player.getCountry()) ? "" : player.getCountry()` seems  useless.
And duplicate code